docs/scos/dev/guidelines/monitorable-process-guidelines.md

```diff
@@ -0,0 +1,245 @@
+            "find_by": []
+        }
+    }
+}
+```
+
+Log structure with error values:
+
+```JSON
+{
+  "@timestamp": "2021-08-19T14:54:23.447685+00:00",
+  "@version": 1,
+  "host": "localhost",
+  "message": "Exception - Sniffer run was not successful: Unknown error in \"/.../vendor/spryker/development/src/Spryker/Zed/Development/Business/ArchitectureSniffer/ArchitectureSniffer.php::165\"",
+  "type": "ZED",
+  "channel": "Zed",
+  "level": "CRITICAL",
+  "monolog_level": 500,
+  "extra": {
+    "environment": {
+      "application": "ZED",
+      "environment": "development",
+      "store": "US",
+      "codeBucket": "US",
+      "locale": "en_US"
+    },
+    "server": {
+      "url": "http://:/",
+      "is_https": false,
+      "hostname": "",
+      "user_agent": null,
+      "user_ip": null,
+      "request_method": "cli",
+      "referer": null
+    },
+    "request": {
+      "requestId": "ad26d9e1",
+      "type": "CLI",
+      "request_params": []
+    }
+  },
+  "context": {
+    "exception": {
+      "class": "Exception",
+      "message": "Sniffer run was not successful: Unknown error",
+      "code": 0,
+      "file": "/.../vendor/spryker/development/src/Spryker/Zed/Development/Business/ArchitectureSniffer/ArchitectureSniffer.php:165",
+      "trace": [
+        "/.../vendor/spryker/development/src/Spryker/Zed/Development/Business/ArchitectureSniffer/ArchitectureSniffer.php:117",
+        "/.../vendor/spryker/development/src/Spryker/Zed/Development/Business/DevelopmentFacade.php:484",
+        "/.../vendor/spryker/development/src/Spryker/Zed/Development/Communication/Console/CodeArchitectureSnifferConsole.php:286",
+        "/.../vendor/spryker/development/src/Spryker/Zed/Development/Communication/Console/CodeArchitectureSnifferConsole.php:93",
+        "/.../vendor/symfony/console/Command/Command.php:258",
+        "/.../vendor/symfony/console/Application.php:938",
+        "/.../vendor/symfony/console/Application.php:266",
+        "/.../vendor/spryker/console/src/Spryker/Zed/Console/Communication/Bootstrap/ConsoleBootstrap.php:111",
+        "/.../vendor/symfony/console/Application.php:142",
+        "/.../vendor/spryker/console/bin/console:27"      
+      ]
+    }
+  }
+}
+```
+
+## Metric generation
+Each metric represents a condition of some system attributes. There can be many of them, and they can be correlated with each other.
+
+* Every service/component should define and generate project-appropriate metrics for key processes (with a basic dimension of the outcome 
+of the operation (success, failure) and duration) to enable tracking of such events and reacting when they reach undesired scores.
+  * The start and end of a process must be recorded to enable tracking and tuning of the infrastructure.
+  * Communication durations with remote services must be recorded to understand whether the local process is delayed for a good reason.
+* Critical metrics, along with their threshold values, must be highlighted in the 
```

Comment:
any examples here ?  Because I tend to separate the metrics of the system (infra, runtime, ressources) from the software (user's behaviour, in/out requests)